test/cppsuite/tests/background_compact.cpp

```diff
@@ -0,0 +1,362 @@
+
+            /* Look for an existing random cursor in our cursor cache. */
+            if (rnd_cursors.find(coll.id) == rnd_cursors.end()) {
+                logger::log_msg(
+                  LOG_TRACE, log_prefix + "Creating cursors for collection: " + coll.name);
+                /* Open the two cursors for the chosen collection. */
+                scoped_cursor rnd_cursor =
+                  tw->session.open_scoped_cursor(coll.name, "next_random=true");
+                rnd_cursors.emplace(coll.id, std::move(rnd_cursor));
+                std::string stat_uri = STATISTICS_URI + coll.name;
+                scoped_cursor stat_cursor = tw->session.open_scoped_cursor(stat_uri);
+                stat_cursors.emplace(coll.id, std::move(stat_cursor));
+            }
+
+            /* Get the cursors associated with the collection. */
+            scoped_cursor &stat_cursor = stat_cursors[coll.id];
+            scoped_cursor &rnd_cursor = rnd_cursors[coll.id];
+
+            /* Get the file statistics so we know how much to truncate. */
+            int64_t entries, bytes_avail_reuse, file_size;
+            metrics_monitor::get_stat(stat_cursor, WT_STAT_DSRC_BTREE_ENTRIES, &entries);
+            metrics_monitor::get_stat(
+              stat_cursor, WT_STAT_DSRC_BLOCK_REUSE_BYTES, &bytes_avail_reuse);
+            metrics_monitor::get_stat(stat_cursor, WT_STAT_DSRC_BLOCK_SIZE, &file_size);
+
+            /* Don't truncate if we already have enough free space for compact to do work. */
+            int64_t pct_free_space = (bytes_avail_reuse * 100 / file_size);
+            if (pct_free_space > 10) {
+                logger::log_msg(LOG_INFO,
+                  log_prefix + "Skip truncating collection {" + std::to_string(coll.id) +
+                    "}, free space available = " + std::to_string(pct_free_space));
+                testutil_check(stat_cursor->reset(stat_cursor.get()));
+                continue;
+            }
+
+            /*
+             * Truncate a range of keys between 0 and 100 until we've truncated a total of 20% of
+             * the entries in the table.
+             */
+            int64_t n_keys_to_truncate = (entries / 100) * 20;
+            int64_t keys_truncated = 0, retries = 0;
+            const uint64_t MAX_RETRIES = 100;
+            while (keys_truncated < n_keys_to_truncate && tw->running() && retries < MAX_RETRIES) {
+                /* Start a transaction if possible. */
+                tw->txn.try_begin();
+
+                /* Choose a random key to delete. */
+                int ret = rnd_cursor->next(rnd_cursor.get());
+
+                if (ret != 0) {
+                    /*
+                     * It is possible not to find anything if the collection is empty. In that case,
+                     * finish the current transaction as we might be able to see new records after
+                     * starting a new one.
+                     */
+                    if (ret == WT_NOTFOUND) {
```

Comment:
Technically our style says no brackets on single line ifs. This applies to a lot of the if statements in this file.

Reply:
Fixed here and in one other place here https://github.com/wiredtiger/wiredtiger/blob/a8fb8d9106aa9d06e60bcd54bece38e5cd51a7f2/test/cppsuite/tests/background_compact.cpp#L265-L274, I don't see any others?